### PR TITLE
fleleshing out remaining rspec tests

### PR DESCRIPTION
**Pull request by dostapenko**

### CODE REVIEW

File: lib/server.rb [hunk 1]
```diff
@@ -58,20 +60,42 @@ def ipn_response(ipn)
 
   def computer_testing(params)
     id = params['my_id']
-    if (params['test_mode'] == 'on')
-      #unless (computer_online?(id))
+    if params['test_mode'] == 'on'
+      if (!computer_online?(id))
```
Comment (dostapenko) [inline lib/server.rb:64]: this is used to make sure that no1 else is online and using that sandbox for testing. If that sandbox is in use, then it goes to the else statement. The else sends emails to both developers and shuts down testing

File: lib/server.rb [hunk 2]
```diff
@@ -142,26 +166,41 @@ def ipn_response_present?(paypal_id)
...
     elsif ipn_response_present?(paypal_id)
       send_verification
     else
-      send_ipn(paypal_id)
+      send_ipn_if_present(paypal_id)
     end
   end
 
-  def unexpected_poll(paypal_id)
+  def unexpected_poll(paypal_id, content=nil)
```
Comment (dostapenko) [inline lib/server.rb:180]: currently, the content param is not used but im trying to combine this method with the one in server_poll_checker. This is where I got stuck today. Im thinking of maybe moving this whole method there. Not sure yet what Im going to do.

File: lib/server.rb [hunk 3]
```diff
@@ -142,26 +166,41 @@ def ipn_response_present?(paypal_id)
...
     @unexpected_poll_time[paypal_id] = Time.now
 
-    to =  @email_map[paypal_id]
-    subject = 'Unexpected poll from your developer machine'
-    body = 'Your computer made an unexpected poll on the Superbox IPN forwarder. The poll occurred before test mode was turned on'
+    unless(@email_map[paypal_id] == nil)
```
Comment (dostapenko) [inline lib/server.rb:183]: this is in case the developers email is unknown. sends to all devs

**Comment by oldfartdeveloper** [comment]
Nice job.  Merging

**Pull request merged**